lib/src/otx/data/entity/sample.py

```
@@ -0,0 +1,118 @@
+from datumaro.components.media import Image
+from datumaro.experimental.dataset import Sample
+from datumaro.experimental.fields import image_field, label_field
+from torchvision import tv_tensors
+
+from otx.data.entity.base import ImageInfo
+
+if TYPE_CHECKING:
+    from datumaro import DatasetItem, Polygon
+    from torchvision.tv_tensors import BoundingBoxes, Mask
+
+
+class OTXSample(Sample):
+    """Base class for OTX data samples."""
+
+    image: np.ndarray | torch.Tensor | tv_tensors.Image | Any
+
+    def as_tv_image(self) -> None:
+        """Convert image to torchvision tv_tensors Image format."""
+        if isinstance(self.image, tv_tensors.Image):
+            return
+        if isinstance(self.image, (np.ndarray, torch.Tensor)):
+            self.image = tv_tensors.Image(self.image)
+            return
+        msg = "OTXSample must have an image"
+        raise ValueError(msg)
+
+    @property
+    def masks(self) -> Mask | None:
+        """Get masks for the sample."""
+        return None
+
+    @property
+    def bboxes(self) -> BoundingBoxes | None:
+        """Get bounding boxes for the sample."""
+        return None
+
+    @property
+    def keypoints(self) -> torch.Tensor | None:
+        """Get keypoints for the sample."""
+        return None
+
+    @property
+    def polygons(self) -> list[Polygon] | None:
+        """Get polygons for the sample."""
+        return None
+
+    @property
+    def label(self) -> torch.Tensor | None:
+        """Optional label property that returns None by default."""
+        return None
+
+    @property
+    def img_info(self) -> ImageInfo | None:
+        """Get image information for the sample."""
+        if getattr(self, "_img_info", None) is None:
+            image = getattr(self, "image", None)
+            if image is not None and hasattr(image, "shape") and len(image.shape) == 3:
+                img_shape = image.shape[:2]
+            else:
+                return None
```

In what situations is this property expected to return None? Wouldn't it be better to raise an exception if it's not possible to extract image information from the sample?

---

It is very hard to understand all usages of OTXDataItem.img_info and if they would work with None or not. For now, it is better to maintain the existing logic until we start to factor out OTXDataItem.